plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -14,14 +19,28 @@ import Codec.CBOR.Decoding qualified as CBOR
 import Codec.CBOR.Extras
 import Codec.CBOR.Read qualified as CBOR
 import Control.Arrow ((>>>))
+import Control.Exception
+import Control.Lens
+import Control.Monad.Error.Lens
 import Control.Monad.Except
-import Data.Bifunctor
-import Data.ByteString.Lazy (fromStrict)
+import Data.ByteString.Lazy as BSL (ByteString, fromStrict)
 import Data.ByteString.Short
 import Data.Coerce
 import Data.Set as Set
 import Prettyprinter
 
+data ScriptDecodeError =
+      CBORDeserialiseError CBOR.DeserialiseFailure
+    | RemainderError BSL.ByteString
+    | LanguageNotAvailableError
+        { sdeAffectedLang :: LedgerPlutusVersion
+        , sdeIntroPv      :: ProtocolVersion
+        , sdeCurrentPv    :: ProtocolVersion
+        }
+    deriving stock (Eq, Show)
+    deriving anyclass Exception
```

Comment:
plutus-benchmark complained about it, because it rethrows to `SomeException`-like exception. https://github.com/input-output-hk/plutus/blob/master/plutus-benchmark/validation/BenchDec.hs#L45